backend/app/store/service/service.go

```diff
@@ -101,7 +101,8 @@ func (s *DataStore) Create(comment store.Comment) (commentID string, err error)
 		comment.PostTitle = title
 	}()
 
-	s.submitImages(comment.Locator, comment.ID)
+	defer s.submitImages(comment) // submit images only after creation in Engine
```

Comment:
I don't really like defer use for this. Probably just adding it directly after Create call will be easier to read and understand.